bottles/frontend/windows/envvars.py

```diff
@@ -36,53 +40,90 @@ def __init__(self, parent, env, **kwargs):
         self.manager = parent.window.manager
         self.config = parent.config
         self.env = env
-
-        self.set_title(self.env[0])
-        self.set_text(self.env[1])
+        self.set_text("=".join(self.env))
 
         # connect signals
+        self.connect("changed", self.__validate)
         self.connect("apply", self.__save)
         self.btn_remove.connect("clicked", self.__remove)
 
+        try:
+            widget = (
+                self.get_child().get_first_child().get_next_sibling().get_first_child()
+            )
+            while isinstance(widget, Gtk.Label):
+                widget.set_visible(False)
+                widget = widget.get_next_sibling()
+
+            if isinstance(widget, Gtk.Text):
+                widget.set_valign(Gtk.Align.CENTER)
+            else:
+                raise RuntimeError("Could not find widget Gtk.Text")
+        except Exception as e:
+            logging.error(
+                f"{type(e)}: {e}\nEnvironmentVariableEntryRow could not find text widget. Did AdwEntryRow change it's widget tree?"
+            )
```

Comment:
I think leaving a comment explaining why this hack is necessary would help future readers

Reply:
Moved to method with docstring.